src/Quartz/Impl/JobType.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+
+using Quartz.Util;
+
+namespace Quartz.Impl;
+
+/// <summary>
+/// Store the Job Type and backing field name for serialization
+/// </summary>
+public class JobType
+{
+    private string storableTypeName = string.Empty;
+    private Lazy<Type> lazyType = new(() => throw new InvalidOperationException("Type not defined"));
+
+    public JobType()
+    {
+    }
+
+    public JobType(string? typeStorageName)
+    {
+        this.StorableTypeName = typeStorageName ?? throw new ArgumentNullException(nameof(typeStorageName));
+    }
+
+    /// <summary>
+    /// Job Type declaration
+    /// </summary>
+    public JobType(Type type)
+    {
+        Type = type;
+    }
+
+    /// <summary>
+    /// Type as a string represented storable name.
+    /// </summary>
+    public string StorableTypeName
```

Comment:
I think this is just `Name`, or `FullName`, should follow same sort of naming logic in `Type`